### ISSUE TITLE
Cannot find method System.Threading.Tasks.Task Audit.Core.AuditScope.DisposeAsync()

### ISSUE THREAD

**Issue by Meteoeoeo** [thepirat000/Audit.NET#371]
**Describe the bug**
In ASP.NET Core Web Application based on .NET Framework I got System.MissingMethodException - cannot find method 'System.Threading.Tasks.Task Audit.Core.AuditScope.DisposeAsync()'

When I download source code and changed return type of method DisposeAsync to ValueTask everythings work correctly(in AuditScope and IAuditScope).

Probably bug is in the use of preprocessor directives.

**To Reproduce**

1. Create new project ASP.NET Core Web Application (.NET Framework + ASP.NET CORE 2.1, Web API)
2. Install Audit.NET + Audit.NET.WebApiCore
3. Add Middleware to Setup.cs:
```
      app.UseAuditMiddleware(x => x
        .WithEventType("{verb}:{url}")
        .IncludeHeaders()
        .IncludeResponseHeaders()
        .IncludeRequestBody()
        .IncludeResponseBody());
```
4. Run application and send request to /api/values - response will be with statuscode 500

**Expected behavior**
Not throw exception :)

**Libraries (specify the Audit.NET extensions being used including version):**
 - Audit.NET.WebApiCore: 16.4.5
 - Audit..NET: 16.4.5

**Target .NET framework:**
I have .NET Core 2.1 application based on .NET Framework 4.7.2

**Additional context**
It was working on Audit.NET version 15.3

**Comment by thepirat000** [comment]
Please share your csproj targetframework and package references

**Comment by Meteoeoeo** [comment]
```
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net472</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Folder Include="wwwroot\" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Audit.NET" Version="16.4.5" />
    <PackageReference Include="Audit.NET.SqlServer" Version="16.4.5" />
    <PackageReference Include="Audit.WebApi.Core" Version="16.4.5" />
    <PackageReference Include="Microsoft.AspNetCore" Version="2.1.7" />
    <PackageReference Include="Microsoft.AspNetCore.HttpsPolicy" Version="2.1.1" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.1.3" />
  </ItemGroup>

</Project>

```

**Comment by thepirat000** [comment]
I will fix this by standardizing the `DisposeAsync()` to return a `ValueTask` for any target framework. Will also need to add the dependency to `System.Threading.Tasks.Extensions` for `net45` and `netstandard1.3` targets.

**Comment by thepirat000** [comment]
This was fixed on version 16.5.0, please upgrade your references and re-test
